dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu.h

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_BRIGHTNESS_CONTRAST_GPU_H_
+#define DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_BRIGHTNESS_CONTRAST_GPU_H_
+
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/core/geom/box.h"
+#include "dali/kernels/common/block_setup.h"
+
+namespace dali {
+namespace kernels {
+namespace brightness_contrast {
+
+
+template <size_t ndims>
+using Roi = Box<ndims, int>;
+
+
+template <class InputType, class OutputType, int ndims>
+struct SampleDescriptor {
+  const InputType *in;
+  OutputType *out;
+  ivec<ndims> in_pitch, out_pitch;
+  float brightness, contrast;
+};
+
+
+template <size_t ndims>
+TensorListShape<ndims> RoiToShape(const std::vector<Roi<ndims>> &rois, int nchannels) {
+  std::vector<TensorShape<ndims>> ret;
+
+  for (auto roi : rois) {
+    assert(all_coords(roi.hi >= roi.lo) && "Cannot create a tensor shape from an invalid Box");
+    TensorShape<ndims> ts;
+    auto e = roi.extent();
+    auto ridx = ndims;
+    ts[--ridx] = e[0] * nchannels;
+    for (size_t idx = 1; idx < ndims; idx++) {
+      ts[--ridx] = e[idx];
+    }
+    ret.emplace_back(ts);
+  }
+
+  return TensorListShape<ndims>{ret};
+}
+
+
+template <size_t ndims>
+std::vector<Roi<ndims>>
+AdjustRois(const std::vector<Roi<ndims>> rois, const TensorListShape<ndims + 1> &shapes) {
+  assert(rois.empty() || rois.size() == static_cast<size_t>(shapes.num_samples()));
+  std::vector<Roi<ndims>> ret(shapes.num_samples());
```

Comment:
maybe
```suggestion
  std::vector<Roi<ndims>> ret;
  ret.reserve(shapes.num_samples());
```
would avoid the default construction and later assignment